.github/workflows/repo-sync.yml

```diff
@@ -1,136 +1,154 @@
+              console.error(`Unable to compute the files difference between ${head} and ${base}`, err.message)
+            }
+
+            console.log('Creating a new pull request')
+            const body = `
+            This is an automated pull request to sync changes between the public and private repos.
+            Our bot will merge this pull request automatically.
+            To preserve continuity across repos, _do not squash_ this pull request.
+            `
+            let pull, pull_number
+            try {
+              const response = await github.rest.pulls.create({
+                owner,
+                repo,
+                head,
+                base,
+                title: 'Repo sync',
+                body,
+              })
+              pull = response.data
+              pull_number = pull.number
+              console.log('Created pull request successfully', pull.html_url)
+            } catch (err) {
+              // Don't error/alert if there's no commits to sync
+              // Don't throw if > 100 pulls with same head_sha issue
+              if (err.message?.includes('No commits') || err.message?.includes('same head_sha')) {
+                console.log(err.message)
+                return
+              }
+              throw err
+            }
+
+            console.log('Locking conversations to prevent spam')
+            try {
+              await github.rest.issues.lock({
+                ...context.repo,
+                issue_number: pull_number,
+                lock_reason: 'spam'
+              })
+              console.log('Locked the pull request to prevent spam')
+            } catch (error) {
+              console.error('Failed to lock the pull request.', error)
+              // Don't fail the workflow
+            }
+
+            console.log('Counting files changed')
+            const { data: prFiles } = await github.rest.pulls.listFiles({ owner, repo, pull_number })
+            if (prFiles.length) {
+              console.log(prFiles.length, 'files have changed')
+            } else {
+              console.log('No files changed, closing')
+              await closePullRequest(pull_number)
+              return
+            }
 
-      # Sometimes after updating the branch, there aren't any remaining files changed.
-      # If not, we should close the PR instead of merging it and triggering deployments.
-      - name: Close the pull request if no files remain
-        if: ${{ steps.find-pull-request.outputs.number && steps.pr-files.outputs.count == '0' }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.PAT_WITH_REPO_AND_WORKFLOW_SCOPE }}
-        run: |
-          gh pr close ${{ steps.find-pull-request.outputs.number }} --repo $GITHUB_REPOSITORY
+            console.log('Checking for merge conflicts')
+            if (pull.mergeable_state === 'dirty') {
+              console.log('Pull request has a conflict', pull.html_url)
+              await closePullRequest(pull_number)
```

Comment:
My only hesitation is here. Do we really want to close the pull request when there is a merge conflict? I think it would be better to:

1. leave the PR up, but comment that it has a merge conflict (Should we be notified when it does fail? I don't completely understand what we use dev-internal for yet.)
2. fail noisily

That way someone can use the existing PR as a starting point for fixing the merge conflict.

Reply:
Sweet, I dig this! I'll remove the `closePullRequest()` so that we leave the PR up, so that it doesn't have to be re-opened 👍 I think the fail will be noisy in that the workflow will fail... If we want to go beyond that (eg notification of some type), I'd submit that we should treat that as separate from the current effort of getting `repo-sync` to run again.